### ISSUE TITLE
Removal of libsyclinterface DPCTLQueueMgr_* functions

### ISSUE THREAD

**Issue by diptorupd** [IntelPython/dpctl#1575]
The `dpctl_sycl_queue_manager` is a legacy of an old design in DPEP stack where we were hoping to use a single Python context manager to define the execution queue globally. DPEP software stack now uses the array API compute follows data programming model, as such the following functions are redundant and should be removed:

> 
> DPCTL_API __dpctl_give [DPCTLSyclQueueRef](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/global.html#doxid-df-d57-dpctl-sycl-types-8h-1af23137a8bf20ff63b10db9f7f987799a)
> [DPCTLQueueMgr_GetCurrentQueue](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/group_QueueManager.html#doxid-d0-dde-group-queue-manager-1gaa1e5044153f900fff9b779b30f3b0538)(void);
> 
> DPCTL_API size_t
> [DPCTLQueueMgr_GetQueueStackSize](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/group_QueueManager.html#doxid-d0-dde-group-queue-manager-1gac4dacf08c61ae001907660867185ea7a)(void);
> 
> DPCTL_API bool
> [DPCTLQueueMgr_GlobalQueueIsCurrent](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/group_QueueManager.html#doxid-d0-dde-group-queue-manager-1ga38b8e0f4ea42601d27eda0d5413c5164)(void);
> 
> DPCTL_API bool
> [DPCTLQueueMgr_IsCurrentQueue](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/group_QueueManager.html#doxid-d0-dde-group-queue-manager-1ga77d39c59561d3f427a1dde1ce81652db)(__dpctl_keep const [DPCTLSyclQueueRef](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/global.html#doxid-df-d57-dpctl-sycl-types-8h-1af23137a8bf20ff63b10db9f7f987799a) QRef);
> 
> DPCTL_API void
> [DPCTLQueueMgr_PopQueue](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/group_QueueManager.html#doxid-d0-dde-group-queue-manager-1gac5fa7690c281328a070f2409b548df73)(void);
> 
> DPCTL_API void
> [DPCTLQueueMgr_PushQueue](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/group_QueueManager.html#doxid-d0-dde-group-queue-manager-1gac9c2b3d22c3bfb0797c0b35551d331b7)(__dpctl_keep const [DPCTLSyclQueueRef](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/global.html#doxid-df-d57-dpctl-sycl-types-8h-1af23137a8bf20ff63b10db9f7f987799a) QRef);
> 
> DPCTL_API void
> [DPCTLQueueMgr_SetGlobalQueue](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/group_QueueManager.html#doxid-d0-dde-group-queue-manager-1gaf767ec0189383390b6bc63835eef1543)(__dpctl_keep const [DPCTLSyclQueueRef](https://intelpython.github.io/dpctl/latest/docfiles/libsyclinterface/global.html#doxid-df-d57-dpctl-sycl-types-8h-1af23137a8bf20ff63b10db9f7f987799a) QRef);
> 
> 

On dpctl's side I believe the whole dpctl._sycl_queue_manager.pyx can be removed.

**Comment by diptorupd** [comment]
@oleksandr-pavlyk @ndgrigorian If you agree, I can create a PR.

**Comment by ndgrigorian** [comment]
> @oleksandr-pavlyk @ndgrigorian If you agree, I can create a PR.

I agree, though dpnp may need to be changed before we can merge the PR. There should be nothing blocking opening a PR in dpctl.

**Comment by oleksandr-pavlyk** [comment]
@diptorupd Resolving this issue would also close #11

**Comment by diptorupd** [comment]
> @diptorupd Resolving this issue would also close #11

Of course, how can I miss that 😄
